### ISSUE TITLE
A way to stop searching when first match is found

### ISSUE THREAD

**Issue by benmccann** [SuperchupuDev/tinyglobby#24]
I saw this requested in https://github.com/webpro-nl/knip/issues/768

**Comment by SuperchupuDev** [comment]
it is possible to add (fdir exposes an option for it), but i'm not sure about adding extra options just because they could be useful for some. i feel like the proper fix would be to add `globStream`

**Comment by benmccann** [comment]
Yeah, I think if you click through to see how it was originally done it was with `globStream`

**Comment by benmccann** [comment]
Immich is another project that we can switch over to tinyglobby once we have `globStream`

**Comment by etnoy** [comment]
Seems like fdir is lacking any such support, however

**Comment by benmccann** [comment]
Here's the issue in `fdir` that this depends on: https://github.com/thecodrr/fdir/issues/40

**Comment by benmccann** [comment]
I wonder if this could be implemented now that `fdir` uses `AbortController`. See:
- https://github.com/SuperchupuDev/tinyglobby/issues/134
- https://github.com/thecodrr/fdir/pull/110
- https://github.com/thecodrr/fdir/pull/144
